src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectInfo.cs

```diff
@@ -618,7 +618,8 @@ internal Type[] GetMemberTypes(string[] inMemberNames, Type objectType)
             bool isFound = false;
             for (int i = 0; i < _cache._memberInfos.Length; i++)
             {
-                if (!memberMissing && inMemberNames[i].Equals(_cache._memberInfos[i].Name))
+                if (!memberMissing && 
+                    inMemberNames[i].Equals(FormatterServices.GetFullMemberInfoName(_cache._memberInfos[i], objectType)))
```

Comment:
My email inbox is currently locked, I already wrote an explanation what is happening but it's hanging around in the Outbox. Here's what I investigated:

This is the place where the field name gets created in netfx: https://referencesource.microsoft.com/#mscorlib/system/runtime/serialization/serializationfieldinfo.cs,47

In core everything is handled in this code block:
https://github.com/dotnet/corefx/blob/c5abd4a0b4c299cb49098f7ebd33ff22e3914568/src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs#L33-L42